docs/content/latest/develop/learn/ttl-data-expiry.md

```diff
@@ -1,30 +1,102 @@
+in master’s syscatalog as part of the table’s schema). If no TTL is present at the column’s value,
+the table TTL acts as the default value.
 
-<ul class="nav nav-tabs nav-tabs-yb">
-  <li>
-    <a href="#cassandra" class="nav-link active" id="cassandra-tab" data-toggle="tab" role="tab" aria-controls="cassandra" aria-selected="true">
-      <i class="icon-java-bold" aria-hidden="true"></i>
-      YCQL
-    </a>
-  </li>
-  <li>
-    <a href="#redis" class="nav-link" id="redis-tab" data-toggle="tab" role="tab" aria-controls="redis" aria-selected="true">
-      <i class="icon-java-bold" aria-hidden="true"></i>
-      YEDIS
-    </a>
-  </li>
-</ul>
-
-<div class="tab-content">
-  <div id="cassandra" class="tab-pane fade show active" role="tabpanel" aria-labelledby="cassandra-tab">
-    {{% includeMarkdown "ycql/ttl-data-expiry.md" /%}}
-  </div>
-  <div id="redis" class="tab-pane fade" role="tabpanel" aria-labelledby="redis-tab">
-    {{% includeMarkdown "yedis/ttl-data-expiry.md" /%}}
-  </div>
-</div>
+Furthermore, YCQL has a distinction between rows created using Insert vs Update. We keep track of
+this difference (and row level TTLs) using a "liveness column", a special system column invisible to
+the user. It is added for inserts, but not updates: making sure the row is present even if all
+non-primary key columns are deleted only in the case of inserts.
+
+ 
+## Table level TTL
+
+YCQL allows the TTL property to be specified at the table level. 
+In this case, we do not store the TTL on a per KV basis in DocDB; but the TTL is implicitly enforced 
+on reads as well as during compactions (to reclaim space).
+Table level TTL can be defined with `default_time_to_live` [property](../../../api/ycql/ddl_create_table#table-properties-1). 
+
+Below, we will look at how the row-level TTL is achieved in detail.
+
+## Row level TTL
+
+YCQL allows the TTL property to be specified at the level of each INSERT/UPDATE operation. 
+Row level TTL expires the whole row. The value is specified at insert/update time with `USING TTL` clause.
+In such cases, the TTL is stored as part of the DocDB value. A simple query would be:
+
+```cassandraql
+INSERT INTO pageviews(path) VALUES ('/index') USING TTL 10;
+SELECT * FROM pageviews;
+
+ path   | views
+--------+-------
+ /index |  null
+
+(1 rows)
+```
+After 10 seconds, the row is expired:
+
+```cassandraql
+SELECT * FROM pageviews;
+
+ path | views
+------+-------
+
+(0 rows)
+```
+
+## Column level TTL
+
+YCQL also allows to have column level TTL independent on each row.  
+In such cases, the TTL is stored as part of the DocDB column value. 
+But we can achieve it only when updating the column:
+
+```cassandraql
+INSERT INTO pageviews(path,views) VALUES ('/index', 10);
+
+SELECT * FROM pageviews;
+
+ path   | views
+--------+-------
+ /index |  10
+
+(1 rows)
+
+UPDATE pageviews USING TTL 10 SET views=10 WHERE path='/index';
+```
+
+After 10 seconds, querying for the rows the `views` column will return `NULL`:
```

Comment:
will return `NULL`:
-->
will return `NULL`, but notice that the row still exists.